canonical-web-and-design/maas-ui
# fix(ui): only show curtin logs for failed deployment

## Done

- Only display the curtin log download option for machines that have failed deployment.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Point your UI to Bolla.
- Visit /MAAS/r/machine/mbfgbq/logs and open the download menu. There should be no curtin logs item.
- Visit /MAAS/r/machine/gtenqm/logs and open the download menu. There should be a curtin logs item.

## Fixes

Fixes: #2499.
